chess/gaviota_native.py

```diff
@@ -0,0 +1,2436 @@
+    dummy = fread32(currentStream)
+
+    blocks = (offset - 40) / 4 - 1
+    n_idx = blocks + 1
+
+    # Input of Indexes 
+    for i in xrange(n_idx):
+        idx = fread32(currentStream)
+        p[i] = idx # reads a 32 bit int, and converts it to index_t 
+
+    if (not currentFilename in Zipinfo):
+        Zipinfo[currentFilename] = ZIPINFO()
+
+    Zipinfo[currentFilename].extraoffset = 0
+    Zipinfo[currentFilename].totalblocks = n_idx
+    Zipinfo[currentFilename].blockindex = p
+
+    return True
+
+def egtb_block_getsize_zipped(currentFilename, block):
+        
+    i = Zipinfo[currentFilename].blockindex[block]
+    j = Zipinfo[currentFilename].blockindex[block + 1]
+
+    return j - i
+
+def egtb_block_park(currentFilename,block, currentStream):
+    i = Zipinfo[currentFilename].blockindex[block]
+    i += Zipinfo[currentFilename].extraoffset
+
+    currentStream.seek(i)
+    return i
+
+EGTB_MAXBLOCKSIZE = 65536
+Buffer_zipped = [-1] * EGTB_MAXBLOCKSIZE
+Buffer_packed = [-1] * EGTB_MAXBLOCKSIZE
+
+# bp:buffer packed to out:distance to mate buffer 
+def  egtb_block_unpack(side, n, bp):
+    return [dtm_unpack(side, i) for i in bp[:n]]
+
+
+#public byte[] Decompress(byte[] inputBytes)
+#    #    MemoryStream newInStream = new MemoryStream(inputBytes);
+
+#    SevenZip.Compression.LZMA.Decoder decoder = new SevenZip.Compression.LZMA.Decoder();
+
+#    newInStream.Seek(0, 0);
+#    MemoryStream newOutStream = new MemoryStream();
+
+#    byte[] properties2 = new byte[5];
+#    if (newInStream.Read(properties2, 0, 5) != 5)
+#        throw (new Exception("input .lzma is too short"));
+#    long outSize = 0;
+#    for (i = 0; i < 8 i++)
+#        #        v = newInStream.ReadByte()
+#        if (v < 0)
+#            throw (new Exception("Can't Read 1"));
+#        outSize |= ((long)(byte)v) << (8 * i);
+#    #    decoder.SetDecoderProperties(properties2);
+
+#    long compressedSize = newInStream.Length - newInStream.Position;
+#    decoder.Code(newInStream, newOutStream, compressedSize, outSize, null);
+
+#    byte[] b = newOutStream.ToArray();
+
+#    return b;
+
+
+
+#
+def split_index(entries_per_block, i, o):
+        
+    n = i / entries_per_block;
+    offset = n * entries_per_block;
+    remainder = i - offset
+    return offset, remainder
+
+
+#bool get_dtm_from_cache(int side, int idx, ref int dtm)
+#    #    offset = 0
+#    remainder = 0
+
+#    split_index(entries_per_block, idx, ref offset, ref remainder);
+
+#    int tableIndex = blockCache.FindIndex(delegate(TableBlock tb)
+#        #        return (tb.key == currentFilename) && (tb.offset == offset) && (tb.side == side);
+#    });
+
+#    dtm = blockCache[tableIndex].pcache[remainder];
+
+#    return true;
+#
+class TableBlock:
+
+    def __init__(self, currentFilename, stm, o):
+        self.key = currentFilename
+        self.side = stm
+        self.offset = o
+        self.pcache = []
+
+    # TODO: not sure this is really useful...
+    def __hash__(self):
```

Comment:
As long as the `blockCache` dictionary does not have TableBlocks as *keys* this is not required.